mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -896,6 +896,135 @@ func TestClientSideEncryptionProse(t *testing.T) {
+				nil,
+			},
+			{
+				// If the client has an unlimited maxPoolSize, and a keyVaultClient is set, the driver should reuse the
+				// client for metadata operations.
+				"deadlock case 6", 0, false, true, 1,
+				[]startedEvent{{"listCollections", "db"}, {"insert", "db"}, {"find", "db"}},
+				[]startedEvent{{"find", "keyvault"}},
+			},
+			{
+				// If the client has an unlimited maxPoolSize, and bypassAutomaticEncryption=true, the driver should
+				// reuse the client for keyVault operations
+				"deadlock case 7", 0, true, false, 1,
+				[]startedEvent{{"find", "db"}, {"find", "keyvault"}},
+				nil,
+			},
+			{
+				// If the client has an unlimited maxPoolSize, bypassAutomaticEncryption=true, and a keyVaultClient is
+				// set, the driver should not create an internal Client.
+				"deadlock case 8", 0, true, true, 1,
+				[]startedEvent{{"find", "db"}},
+				[]startedEvent{{"find", "keyvault"}},
+			},
+		}
+
+		for _, tc := range testcases {
+			mt.Run(tc.description, func(mt *mtest.T) {
+				var clientEncryptedEvents []startedEvent
+				var clientEncryptedTopologyOpening int
+
+				d := newDeadlockTest(mt)
+				defer d.disconnect(mt)
+
+				kmsProviders := map[string]map[string]interface{}{
+					"local": {"key": localMasterKey},
+				}
+				aeOpts := options.AutoEncryption()
+				aeOpts.SetKeyVaultNamespace("keyvault.datakeys").
+					SetKmsProviders(kmsProviders).
+					SetBypassAutoEncryption(tc.bypassAutoEncryption)
+				if tc.keyVaultClientSet {
+					aeOpts.SetKeyVaultClientOptions(d.clientKeyVaultOpts)
+				}
+
+				ceOpts := options.Client().ApplyURI(mtest.ClusterURI()).
+					SetMonitor(&event.CommandMonitor{
+						Started: func(ctx context.Context, event *event.CommandStartedEvent) {
+							clientEncryptedEvents = append(clientEncryptedEvents, startedEvent{event.CommandName, event.DatabaseName})
+						},
+					}).
+					SetServerMonitor(&event.ServerMonitor{
+						TopologyOpening: func(event *event.TopologyOpeningEvent) {
+							clientEncryptedTopologyOpening++
+						},
+					}).
+					SetMaxPoolSize(tc.maxPoolSize).
+					SetAutoEncryptionOptions(aeOpts)
+
+				clientEncrypted, err := mongo.Connect(mtest.Background, ceOpts)
+				defer clientEncrypted.Disconnect(mtest.Background)
+				assert.Nil(mt, err, "Connect error: %v", err)
+
+				coll := clientEncrypted.Database("db").Collection("coll")
+				if !tc.bypassAutoEncryption {
+					_, err = coll.InsertOne(mtest.Background, bson.M{"_id": 0, "encrypted": "string0"})
+				} else {
+					unencryptedColl := d.clientTest.Database("db").Collection("coll")
+					_, err = unencryptedColl.InsertOne(mtest.Background, bson.M{"_id": 0, "encrypted": d.ciphertext})
+				}
+				assert.Nil(mt, err, "InsertOne error: %v", err)
+
+				raw, err := coll.FindOne(mtest.Background, bson.M{"_id": 0}).DecodeBytes()
+				assert.Nil(mt, err, "FindOne error: %v", err)
+
+				expected := bsoncore.NewDocumentBuilder().AppendInt32("_id", 0).AppendString("encrypted", "string0").Build()
```

Comment:
nit: I generally find it easier to read if these are split over multiple lines where each `Append` call is a separate line and `Build()` is its own line at the end.

Reply:
Done.